node/consensus/src/lib.rs

```diff
@@ -89,6 +94,80 @@ impl<N: Network> Default for TransactionsQueue<N> {
     }
 }
 
+impl<N: Network> TransactionsQueue<N> {
+    fn contains(&self, transaction_id: &N::TransactionID) -> bool {
+        self.deployments.contains(transaction_id) || self.executions.contains(transaction_id)
+    }
+
+    fn insert(&mut self, transaction_id: N::TransactionID, transaction: Transaction<N>) {
+        if transaction.is_deploy() {
+            self.deployments.put(transaction_id, transaction);
+        } else {
+            self.executions.put(transaction_id, transaction);
+        }
+    }
+}
+
+struct PriorityQueueInner<N: Network> {
+    /// A counter to ensure fifo ordering for transmissions with the same fee.
+    counter: u64,
+    /// A map of transmissions ordered by fee and by fifo sequence.
+    transaction_ids: BTreeMap<(Reverse<U64<N>>, u64), N::TransactionID>,
+    /// A map of transmission IDs to transmissions.
+    transactions: HashMap<N::TransactionID, Transaction<N>>,
```

Comment:
Is it possible to use ((priority_fee, txid) transaction) directly in the BTreeMap?

Then we only need a single container and we can get rid of this wrapper.

I'm assuming the difference in retrieval speed is negligible (walking down the tree v.s. hashing), its just a bit simpler.

Reply:
We could serialize into a string, but yeah at that point the performance comparison becomes unclear, we can leave it as is, thx for the pushback